modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/select/SelectDDMFormFieldTemplateContextContributor.java

```diff
@@ -140,7 +158,20 @@ protected List<Map<String, String>> getOptions(
 		List<Map<String, String>> objectFieldOptions = _getObjectFieldOptions(
 			ddmFormField, ddmFormFieldOptions, ddmFormFieldRenderingContext);
 
+		boolean alphabeticalOrder = GetterUtil.getBoolean(
+			ddmFormField.getProperty("alphabeticalOrder"));
+
 		if (ListUtil.isNotEmpty(objectFieldOptions)) {
+			String language = ServiceContextThreadLocal.getServiceContext(
+			).getLanguageId();
+
+			Locale newLocale = LocaleUtil.fromLanguageId(language);
+
+			if (alphabeticalOrder && (locale != newLocale)) {
+				objectFieldOptions = alphabeticalOrder(
+					objectFieldOptions, newLocale);
+			}
```

Comment:
@vngoliveira please try to use the correct locale [here](https://github.com/liferay-forms/liferay-portal/blob/29601e8f54f470ee6f2984aa284241cbc8f90a81/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/select/SelectDDMFormFieldTemplateContextContributor.java#L269-L270) instead to avoid reordering more than once.

Thanks!

Reply:
@carolmariaabb the correct locale is already set in the parameter

![Screenshot from 2023-10-20 10-13-05](https://github.com/liferay-forms/liferay-portal/assets/92388363/77561e7a-bbda-4030-8eaa-ba6f17284020)

Reply:
@carolmariaabb because when the name has translation, the ordering follows the "default" locale set before, and not the new. So I guess that is necessary to reorder by the locale. I didn't find another way to fix it, sorry.

Reply:
@vngoliveira I still don't understand why we are getting the options labels using a locale [here](https://github.com/liferay-forms/liferay-portal/blob/29601e8f54f470ee6f2984aa284241cbc8f90a81/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/select/SelectDDMFormFieldTemplateContextContributor.java#L289) (after sorting by this same locale [here](https://github.com/liferay-forms/liferay-portal/blob/29601e8f54f470ee6f2984aa284241cbc8f90a81/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/select/SelectDDMFormFieldTemplateContextContributor.java#L269)), and then we need to sort by a different locale.